OracleTuxedo/art_runtime/dockerfiles/buildDockerImage.sh

```diff
@@ -0,0 +1,168 @@
+ORACLE_PRECOMP=oracle-instantclient12.2-precomp-12.2.0.1.0-1.x86_64.rpm
+PATCH_LIST=
+
+while getopts "h:c:r:p:v:" optname; do
+  case "$optname" in
+    "h")
+      usage
+      ;;
+    "c")
+      CIT_INSTALLER="$OPTARG"
+      ;;
+    "r")
+      ARTRT_INSTALLER="$OPTARG"
+      ;;
+    "p")
+      PATCH_LIST="$OPTARG"
+      ;;
+    "v")
+      VERSION="$OPTARG"
+      ;;
+    *)
+    # Should not occur
+      echo "Unknown error while processing options inside buildDockerImage.sh"
+      ;;
+  esac
+done
+
+if [ -z "$VERSION" ]; then   # -v option is a MUST
+    echo "You must specify Tuxedo version with -v option"
+    exit
+fi
+
+if [ ! -e bin/${CIT_INSTALLER} ]
+then
+    echo "Download the Cobol-IT Distribution and"
+    echo "drop the file ${CIT_INSTALLER} in ./bin/ folder before"
+    echo "building this Tuxedo ART Docker container!"
+    exit
+fi
+
+if [ ! -e bin/${ORACLE_BASE} ] && [ ! -e bin/${ORACLE_SQLPLUS} ] && \
+   [ ! -e bin/${ORACLE_DEVEL} ] && [ ! -e ${ORACLE_PRECOMP} ]; then
+  echo "Download the Oracle client ZIP Distribution and"
+  echo "drop the file ${ORACLE_BASE} "
+  echo "              ${ORACLE_SQLPLUS}"
+  echo "              ${ORACLE_DEVEL}"
+  echo "              ${ORACLE_PRECOMP} in ./bin/ folder before"
+  echo "building this Tuxedo ART Docker container!"
+  exit
+fi
+
+if [ ! -z "$PATCH_LIST" ]; then   # -p option is an OPTIONAL
+    PATCH_FILE="" 
+    PATCH_FILE_LIST="" 
+    OLD_IFS="$IFS" 
+    IFS="," 
+    arrPatch=($PATCH_LIST)
+    for s in ${arrPatch[@]}
+    do
+        if [ ! -e bin/${s} ]
+        then
+            echo "Download the patch file and"
+            echo "drop the file ${s} in ./bin/ folder before"
+            echo "building this Tuxedo ART Docker container!"
+            exit
+        fi
+        PATCH_FILE=${PATCH_FILE}"bin/$s "
+        PATCH_FILE_LIST=${PATCH_FILE_LIST}"$s "
+    done
+    IFS="$OLD_IFS"
+fi
+
+echo "====================="
+
+
+if [ ! -e bin/${ARTRT_INSTALLER} ]
+then
+  echo "Download the Tuxedo ART Runtime ZIP Distribution and"
+  echo "drop the file ${ARTRT_INSTALLER} in ./bin/ folder before"
+  echo "building this Tuxedo ART Docker container!"
+  exit 
+fi
+
+echo "====================="
+# Proxy settings
+PROXY_SETTINGS=""
+if [ "${http_proxy}" != "" ]; then
+  PROXY_SETTINGS="$PROXY_SETTINGS --build-arg http_proxy=${http_proxy}"
+fi
+
+if [ "${https_proxy}" != "" ]; then
+  PROXY_SETTINGS="$PROXY_SETTINGS --build-arg https_proxy=${https_proxy}"
+fi
+
+if [ "${ftp_proxy}" != "" ]; then
+  PROXY_SETTINGS="$PROXY_SETTINGS --build-arg ftp_proxy=${ftp_proxy}"
+fi
+
+if [ "${no_proxy}" != "" ]; then
+  PROXY_SETTINGS="$PROXY_SETTINGS --build-arg no_proxy=${no_proxy}"
+fi
+
+if [ "$PROXY_SETTINGS" != "" ]; then
+  echo "Proxy settings were found and will be used during build."
+fi
+
+
+
+# Fix up the locations of things
+cp ${VERSION}/*.rsp ${VERSION}/init.sh .
+sed -e "s:@ARTRT_PKG@:${ARTRT_INSTALLER}:g" \
```

Comment:
You cannot assume the user will use `buildDockerImage.sh` to build the image. Your `Dockerfile` must work unmodified by any external script so that it can be used by automation and orchestration engines.